### DISCUSSION TITLE
How to test if a nested object contains an array with a specific key value pair?

### DISCUSSION THREAD

**Discussion by alewolf** [cypress-io/cypress#15936] [Questions and Help] [answered]
Hi

I figured out how to wait until a nested object exists, and then continue with the test. 

```js
let o = {
    foo: {
        bar: "a"
    }
};
```

```js
cy.window()
    .its('foo.bar')

```

Or test for the value:

```js
cy.window()
    .its('foo.bar')
    .should('eq', 'a')
```

But, it becomes more complicated when a nested object contains an array, and I want to find out if one of those array items contains a specific `id`.

```
let o = {
    foo: {
        bar: [
            {id:'a'},
            {id:'b'}
        ]
    }
};
```

Any idea how I could find out if in above example the `bar` array contains an item with a key equals `b` ?

**Comment by bahmutov** [comment, marked as answer]
I have updated the example for `.its` command https://glebbahmutov.com/cypress-examples/7.0.1/commands/connectors.html#its
and added a recipe showing how to find an object with one or all known properties https://glebbahmutov.com/cypress-examples/7.0.1/recipes/find-object.html 

Hope it helps

It would be nice to have a better solution for checking if a partial object exists in the array, I agree.

**Reply by alewolf** [reply]
You're the best. Thanks!

Generally I love Cypress. But I am constantly looking for better examples for such advanced assertions and it is difficult with the current documentation for a beginner like me to create or find those kind of solutions as the one you showed me.

**Reply by bahmutov** [reply]
I agree, I suggest looking at:
- https://on.cypress.io/assertions
- https://on.cypress.io/should
- https://glebbahmutov.com/cypress-examples/7.0.1/commands/assertions.html
